dfusion_rust_core/src/models/account_state.rs

```diff
@@ -66,25 +68,40 @@ impl AccountState {
         self.state_hash = self.rolling_hash(self.state_index.low_u32());
         valid_withdraws
     }
+
+    pub fn apply_auction(&mut self, orders: &[Order], results: Solution) {
```

Comment:
This could have probably been its own PR.

Reply:
Yes, I agree.